tests/workflow_builder_test.py

```diff
@@ -628,3 +628,100 @@ def test_dax_test_exclude_nodes_on_saga(tmp_path):
                     )
                 else:
                     assert False
+
+
+def test_dax_with_job_in_container(tmp_path):
+    workflow_params = Parameters.from_mapping(
+        {
+            "workflow_name": "Test",
+            "workflow_created": "Testing",
+            "workflow_log_dir": str(tmp_path / "log"),
+            "workflow_directory": str(tmp_path / "working"),
+            "site": "saga",
+            "namespace": "test",
+            "partition": "gaia",
+            "experiment_name": "fred",
+            "home_dir": str(tmp_path),
+        }
+    )
+
+    slurm_params = Parameters.from_mapping(
+        {"partition": "gaia", "num_cpus": 1, "num_gpus": 0, "memory": "4G"}
+    )
+
+    multiply_input_file = tmp_path / "raw_nums.txt"
+    random = Random()
+    random.seed(0)
+    nums = immutableset(int(random.random() * 100) for _ in range(25))
+    multiply_output_file = tmp_path / "multiplied_nums.txt"
+    sorted_output_file = tmp_path / "sorted_nums.txt"
+
+    with multiply_input_file.open("w") as mult_file:
+        mult_file.writelines(f"{num}\n" for num in nums)
+
+    multiply_params = Parameters.from_mapping(
+        {"input_file": multiply_input_file, "output_file": multiply_output_file, "x": 4}
+    )
+    sort_params = Parameters.from_mapping(
+        {"input_file": multiply_output_file, "output_file": sorted_output_file}
+    )
+
+    resources = SlurmResourceRequest.from_parameters(slurm_params)
+    workflow_builder = WorkflowBuilder.from_parameters(workflow_params)
+
+    # Add Container
+    example_docker = workflow_builder.add_container(
+        "example_container", "docker", tmp_path / "docker.img"
```

Comment:
@danielnapierski Pegasus' [Container Execution Model](https://pegasus.isi.edu/docs/5.0.0/user-guide/containers.html#container-execution-model) is detailed here. My understanding is Pegasus pulls the docker image into the directory which the job is going to run in then the docker container is run from that directory location. 

If we give Pegasus a local file path it will use that tar as the docker image otherwise we can configure a remote site to download the image from.